rules:
- id: unchecked-subprocess-call
  patterns:
  - pattern-either:
    - pattern: |
        subprocess.call(...)
  - pattern-not-inside: |
      $S = subprocess.call(...)
  - pattern-not-inside: |
      subprocess.call(...) == $X
  message: |
    This is not checking the return value of this subprocess call; if it fails no exception will be raised. Consider subprocess.check_call() instead
  fix: |
    subprocess.check_call(...)
  severity: WARNING
  languages:
  - python
